src/query/storages/fuse/src/operations/common/processors/transform_serialize_block.rs

```diff
@@ -74,8 +76,27 @@ impl TransformSerializeBlock {
         output: Arc<OutputPort>,
         table: &FuseTable,
         cluster_stats_gen: ClusterStatsGenerator,
+        is_append: bool,
     ) -> Result<Self> {
-        let source_schema = Arc::new(table.schema_with_stream().remove_virtual_computed_fields());
+        // remove virtual computed fields.
+        let mut fields = table
+            .schema()
+            .fields()
+            .iter()
+            .filter(|f| !matches!(f.computed_expr(), Some(ComputedExpr::Virtual(_))))
+            .cloned()
+            .collect::<Vec<_>>();
+        if !is_append {
```

Comment:
Why `!is_append` ?
The bool of `is_append` a bit confusing, possible to change it to a enum for easy understand?